src/resolvers/git.cr

```diff
@@ -100,12 +100,26 @@ module Shards
       "git"
     end
 
+    private KNOWN_PROVIDERS = {"www.github.com", "github.com", "www.bitbucket.com", "bitbucket.com", "www.gitlab.com", "gitlab.com"}
+
     def self.normalize_key_source(key : String, source : String) : {String, String}
       case key
       when "git"
-        {"git", source}
+        uri = URI.parse(source)
+        downcased_host = uri.host.try &.downcase
+        scheme = uri.scheme.try &.downcase
+        if scheme.in?("git", "http", "https") && downcased_host && downcased_host.in?(KNOWN_PROVIDERS)
```

Comment:
```suggestion
        if scheme.in?("git", "http", "https") && downcased_host.in?(KNOWN_PROVIDERS)
```

Reply:
I want to ensure `downcased_host` is not nil here as otherwise I would have to replace `downcased_host.lchop("www.")` with `downcased_host.try &.lchop("www.")`